src/components/edit-form/edit-form.tsx

```diff
@@ -1,49 +1,170 @@
+import { ThunkDispatch } from 'redux-thunk';
+import { AnyAction } from 'redux';
 import Button from '../button/button';
 import { genres } from '../main/info';
-
+import AppContext from '../app-context/app-context';
+import { FormInput } from '../../types/form-input';
 import './style.css';
+import postMovie from '../../state/post-movie';
+import RootState from '../../types/root-state';
+import putMovie from '../../state/put-movie';
+
+type EditFormProps = {
+  onModalClose: () => void;
+};
+
+export default function EditForm({ onModalClose }: EditFormProps) {
+  const { handleCongratModalOpen, movie } = useContext(AppContext);
+
+  const dispatch: ThunkDispatch<RootState, unknown, AnyAction> = useDispatch();
+
+  const { sort, filter, offset } = useSelector(
+    (state: RootState) => state.movieState
+  );
+
+  const params: FormInput | object = movie
+    ? {
+        id: movie.id,
+        title: movie.title,
+        release_date: movie.release_date,
+        genres: movie.genres,
+        runtime: movie.runtime,
+        poster_path: movie.poster_path,
+        vote_average: movie.vote_average,
+        overview: movie.overview,
+      }
+    : {};
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormInput>({
+    defaultValues: params,
+  });
+
+  const onSubmit: SubmitHandler<FormInput> = (data: FormInput) => {
+    if (!movie) {
+      dispatch(
+        postMovie(
+          data,
+          onModalClose,
+          handleCongratModalOpen,
+          filter,
+          sort,
+          offset
+        )
+      );
+    } else {
+      dispatch(putMovie(data, onModalClose, sort));
+    }
+  };
+
+  const requiredMessage = 'This is required';
 
-export default function EditForm() {
   return (
-    <form className="modal-edit-form">
+    <form className="modal-edit-form" onSubmit={handleSubmit(onSubmit)}>
       <label>
         title
-        <input placeholder="Select Title" />
+        <input
+          className={errors.title?.message ? 'modal-input-error' : ''}
+          placeholder="Select Title"
+          inputMode="text"
+          {...register('title', { required: requiredMessage })}
+        />
+        <p className="modal-submit-error">{errors.title?.message}</p>
       </label>
       <label>
         release date
-        <input type="date" placeholder="Select Date" />
-      </label>
-      <label>
-        movie url
-        <input placeholder="https://" />
-      </label>
-      <label>
-        rating
-        <input placeholder="7.8" />
+        <input
+          className={errors.release_date?.message ? 'modal-input-error' : ''}
+          type="date"
+          placeholder="Select Date"
+          {...register('release_date', { required: requiredMessage })}
+        />
+        <p className="modal-submit-error">{errors.release_date?.message}</p>
       </label>
       <label>
         genre
-        <select>
+        <select
+          className={`modal-genre-select ${
```

Comment:
You can use some library for applying multiple classes, for example, clsx or classNames.